### DISCUSSION TITLE
2D cylinder case

### DISCUSSION THREAD

**Discussion by AyushBoral** [xcompact3d/Incompact3d#228] [General]
How do we run the 2D cylinder case? In the input_DNS_2D file, the Lz is taken as 3.14. I did not understand this.
Thank you
Ayush

**Comment by slaizet** [comment]
Hi Ayush,
Oops yes looks like there is an issue with the input file for the 2D cylinder case (it looks like it is the input file for the Taylor Green case). Let me have a look and I will update the file accordingly.
Cheers,
Sylvain

**Reply by AyushBoral** [reply]
Thanks!

**Reply by slaizet** [reply]
All done for you, just ran the simulation for the new input file and it is now working fine.
Cheers,
Sylvain

**Reply by AyushBoral** [reply]
Can you please guide me on how to choose the p_row and p_col for this 2D case because for me is showing an invalid 2D processor grid.

**Comment by slaizet** [comment]
You want p_row and p_col to be multiple of nxm, nym, nzm. 

(if nx=33, then nxm=32 so you can have p_row & p_col equal to 32,16, 8, 4, 2 or 1)

Ideally you want to be as close as p_row=p_col for optimal performance.

Best,

Sylvain

**Reply by martinmw13** [reply]
Greetings,
I am encountering an issue when running this case, specifically with the configuration of p_row and p_col. My simulation utilizes a grid with nx=257 , ny=128 and nz=1, and regardless of the values I assign to p_row and p_col, I receive the following error: "Invalid 2D processor grid. Make sure that min(ny,nz) >= p_col." Any insights or suggestions  would be greatly appreciated. 

Update on the error I'm encountering while running in single-process mode. Currently, the simulation is configured with p_row=1 and p_col=1. The error message I'm receiving is as follows:

"In auto-tuning mode......
p_row x p_col 1 1
WARNING: current 2D DECOMP set-up might not work
Initializing variables...
double free or corruption (out)

Program received signal SIGABRT: Process abort signal.

**Reply by martinmw13** [reply]
I encountered a build error while compiling my code. After uninstalling and reinstalling the program, I ran make clean, make and received the following error:

```fortran [...]
decomp2d/io.f90:1134:13:
Error: Function 'is_contiguous' at (1) has no IMPLICIT type
decomp2d/io.f90:235:13:
     if (.not. is_contiguous(var)) call decomp_2d_abort(-1, "read_one_real : argument must be contiguous")
             1
Error: Function 'is_contiguous' at (1) has no IMPLICIT type
make: *** [Makefile:110: decomp2d/io.o] Error 1

**Reply by slaizet** [reply]
Hi, 

I just tried to run the 2D cylinder case with 1 MPI process and it is working fine (GNU compiler 9.4), using `input_DNS300_LR_2D.i3d`. 

Regarding the compilation bug, please provide all the information (compiler version, MPI library, etc.) so we can try to reproduce the issue.

Best,

Sylvain

**Reply by pbartholomew08** [reply]
This particular compiler error implies an older version of `gfortran` that doesn't support the `f2008` standard.

**Reply by martinmw13** [reply]
Yep, that was it. Thank you both very much for your help!

**Comment by martinmw13** [comment]
Hello, I'm validating this case with the same reference data (Mittal) and I'm getting discrepancies at the cylinder height for the u mean. I'll attach the 257x128 and 401x260 grid results, despite the refinement, both results are nearly identical. Any insights of how to achieve better results or if it's not adequate to compare both data. Thank you for the time!
![umean](https://github.com/xcompact3d/Incompact3d/assets/57778967/15d69f57-5a10-4396-943c-c07f8f7663ae)
![umean](https://github.com/xcompact3d/Incompact3d/assets/57778967/79659274-18bf-4ed7-a800-282fac0d90d0)

**Reply by slaizet** [reply]
Hi,
What is your resolution and domain size in the Z-direction? Or are you trying to compare a 2D flow with a 3D flow?
Best,
Sylvain
